modules/custom-status/custom-status.php

```diff
@@ -763,19 +793,42 @@ public function get_custom_statuses(): array {
 		}, $statuses );
 
 		// Set the internal object cache
-		$this->custom_statuses_cache = $statuses;
+		self::$custom_statuses_cache = $statuses;
 
 		return $statuses;
 	}
 
+	/**
+	 * Get core's 'draft' and 'pending' post statuses, but include our special attributes
+	 *
+	 * @return array
+	 */
+	private static function get_core_post_statuses(): array {
```

Comment:
Similar point as some other comments, but I'm not sure why we need "core" post statuses, any more than we need core EM fields. If we have 0 or 1 statuses, it would be great if we could handle that gracefully in code. Don't need to do that in this PR, just a general thought.

Reply:
Actually, this is something that we should remove entirely. If the post type is unsupported, none of our code should load which means we shouldn't even be doing anything.

Reply:
I added this back in because without this, the sidebar fails to load. We need to issue that initial status change into draft, after which we can let Gutenberg take over. This is because initially the post type is not available as it's a clean new post.

Reply:
This is also why I added pending review as a restricted status because technically that and draft are required by default in core. Us allowing that to be renamed or altered is going to mean extra logic on our end that we need to handle. That is,

- Pending review can be renamed
- But, the slug itself won't be and will stay the same

Instead, for now I think we should just leave pending review and draft as statuses that cannot be altered or touched in any way. Not saying we don't do it mind you just saying for now we do that and revisit it.